invoices/settlement_interceptor.go

```diff
@@ -0,0 +1,267 @@
+package invoices
+
+import (
+	"fmt"
+	"sync"
+
+	"github.com/lightningnetwork/lnd/fn"
+	"github.com/lightningnetwork/lnd/lntypes"
+)
+
+// SafeCallback is a thread safe wrapper around an InterceptorClientCallback.
+type SafeCallback struct {
+	// mu is a mutex that protects the callback field.
+	mu sync.Mutex
```

Comment:
Could also maybe use: https://pkg.go.dev/sync/atomic#Pointer? 

Not sure it fits better tho. Still getting through the diff.

Reply:
Would you like me to change this?